ansible/roles/node_exporter_nginx/tasks/create-certs.yml

```diff
@@ -0,0 +1,72 @@
+---
+- name: Ensure openssl is installed
+  apt:
+    name: "openssl"
+    state: present
+
+- name: Create node_exporter private dir
+  file:
+    path: "{{ node_exporter_private_path}}"
+    state: directory
+    owner: "{{ node_exporter_user }}"
+    group: "{{ node_exporter_group }}"
+    mode: 0700
+
+- name: Generate host private key
+  command: openssl genrsa -out {{ node_exporter_ssl_key_path }} 4096 creates={{ node_exporter_ssl_key_path }}
+
+- name: Set private key permissions
+  file:
+    path: "{{ node_exporter_ssl_key_path }}"
+    owner: "{{ node_exporter_user }}"
+    group: "{{ node_exporter_group }}"
+    mode: "0700"
+
+- name: Generate CSR request
+  command: openssl req -new -sha256 -subj "{{ node_exporter_ssl_cert_subj }}" -key {{ node_exporter_ssl_key_path }} -out {{ node_exporter_ssl_csr_path }} creates={{ node_exporter_ssl_csr_path }}
+
+- name: Set CSR permissions
+  file:
+    path: "{{ node_exporter_ssl_csr_path }}"
+    owner: "{{ node_exporter_user }}"
+    group: "{{ node_exporter_group }}"
+    mode: "0700"
+
+- name: write the root ca private key to temporary directory
+  copy:
+    content: "{{ vault_prometheus_root_ca_key }}"
+    dest: "{{ node_exporter_private_path }}/root_ca.key"
```

Comment:
FYI: `copy` supports `mode` option as well (maybe I'm wrong and it's broken when `content` is used).

But I still suggest signing CSR locally at controller host.